ui/src/pages/backoffice/Patron/PatronDetails/components/ItemsCheckout/state/actions.js

```diff
@@ -26,12 +30,22 @@ export const checkoutItem = (
         type: SUCCESS,
         payload: response.data,
       });
-      dispatch(
-        sendSuccessNotification(
-          'Loan Created',
-          'The new loan was successfully created.'
-        )
-      );
+      dispatch({
+        type: CLEAR_SEARCH,
+      });
+      // put the current loans into loading state until ES updates
```

Comment:
a bit of explanation here, with the approach I think we should use in general
all the states changes are managed here, please notice that it happens also for the action types for other components. since we have a state, I don't think we should pass the action functions around to the components but if it is possible to trigger state from here.
In this case I am checking out the item for the current patron:
1. When I have success response, I clear the whole search
2. Then I say to my current loans that it should go to isLoading state (user knows that it is being processed currently)
3. After the delay from elastic search I update everything, current loans table and I send the notification that it is done

when i have the error I leave everything (search results and query as it was) and I display the error

Reply:
Can you please explain this by voice when you have time?

Reply:
sure!